ideal_device/io/tlb_manager.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+/*
+
+Questions:
+
+- Do we need per arch TLB manager (for example for UC/WC distribution)?
+
+- Do we need to have index in TLBWindow in order to return it properly?
+
+*/
+
+#include "pci_device.h"
+#include "tlb_window.h"
+
+namespace tt::umd {
+
+class TLBManager {
+
+public:
+    virtual TLBWindow get_tlb_window(tt_xy_pair core) = 0;
+    
+    virtual TLBWindow get_wc_tlb_window() = 0;
+
+    virtual TLBWindow get_uc_tlb_window() = 0;
+
+    virtual void release_tlb_window(TLBWindow& tlb_window) = 0;
+
+    // This is something that is probably going to need per arch implementation.
+    virtual TLBWindow get_dram_tlb_widow(uint32_t dram_channel, uint32_t addr = 0) = 0;
+
+    virtual TLBWindow release_dram_tlb_window(TLBWindow& tlb_window) = 0;
+protected:
+    // Are these constants supposed to be in per arch tlb manager class?
+    static constexpr uint32_t uc_window_start = 0;
+    static constexpr uint32_t uc_window_count = 0;
+
+    static constexpr uint32_t wc_window_start = 0;
+    static constexpr uint32_t wc_window_count = 0;
+
+    // dummy implementation
+    uint32_t map_core_to_tlb(tt_xy_pair core) {
+        return 0;
+    }
+}
+
+// This should represent how we do things today. We have whole BAR0/4 space
+// and just map the TLB we need.
+class StandardTLBManager : public TLBManager {
+
+public:
+    StandardTLBManager(PCIDevice* pci_device) : pci_device(pci_device) {}
+
+    TLBWindow get_tlb_window(tt_xy_pair core) override {
+        uint32_t tlb_index = map_core_to_tlb_index(core);
```

Comment:
So this is actually pci_device-> map_core_to_tlb_index
This is behind which call the architecture specific stuff is hidden (like what exact bar address is used to configure tlbs)?

Reply:
Yes, makes sense to have the mapping in TTDevice or PCIDevice, wherever our soc descriptor is going to live